doc/source/releases.rst

```diff
@@ -0,0 +1,203 @@
+.. _release:
+
+*************
+Release Notes
+*************
+
+This is the list of changes to pandas between each release. For full details,
+see the commit logs at http://github.com/pandas-dev/pandas. For install and
+upgrade instructions, see :ref:`install`.
+
+0.24 release
+------------
```

Comment:
I think it would look better (also looking at your render from the OP) to have
```diff
+ Version 0.24
- 0.24 release
  ------------
```

etc.